packages/editor/src/components/post-fields/index.ts

```diff
@@ -28,29 +24,44 @@ interface Author {
 }
 
 function usePostFields(): UsePostFieldsReturn {
+	const postType = 'page'; // TODO: this could be page or post (experimental).
```

Comment:
We already have two post types that work with the same fields: the `Pages` page pulls data for the `page` post type, and the experimental post screen (enable in "Gutenberg > Experiments > Redesigned post dashboard") does the same with posts.

Reply:
While they have the same fields in dataviews, this is how we display them in the post editor:

| Pages | Posts |
| --- | --- |
| <img width="290" alt="Screenshot 2024-11-20 at 18 34 47" src="https://github.com/user-attachments/assets/09c0f153-2e67-4412-b45a-d7d6d54f9e5b"> | <img width="292" alt="Screenshot 2024-11-20 at 18 34 32" src="https://github.com/user-attachments/assets/3bed0f9e-86b4-45d2-882c-5cfbe590042c"> |

It sounds a follow-up to this PR should be implementing the format field for the `post` post type, and add the `postType` parameter to the `usePostFields` hook.

Reply:
I wonder if ultimately these hooks should received kind/name just like the registration function. I think we'll have this use-case pretty quickly when we start working on the tags/categories dataviews.

Reply:
So, we'd have:

- `useFields( 'postType', 'pages' )` instead of `usePostFields( 'page' )`
- `useFields( 'taxonomy', 'category' )` instead of `useTaxonomyFields( 'category' )`

Under the hood and conceptually, fields and actions are already connected to core-data. Making the `kind` an argument could work.

Reply:
Given we would also use this for taxonomies would it make sense to move these actions/selectors to the `core` data store instead? Given this is more entity specific.
Or in general do we expect every place that uses DataViews/DataForms within WordPress to also import the `@wordpress/editor` package?

Reply:
[This conversation](https://github.com/WordPress/gutenberg/pull/67024#discussion_r1843424806) is relevant: plan is to move them to `@wordpress/fields` package when it becomes stable.